packages/net/tcp_connection.pony

```diff
@@ -648,9 +646,6 @@ actor TCPConnection
           end
           @pony_os_socket_close(fd)
           _try_shutdown()
-          if _connect_count == 0 then
-            _notify.connect_failed(this)
-          end
```

Comment:
this is the source of failure. definitely not what we want given the existence of the happy eyeballs algo.

Reply:
Hmm, I observed that _try_shutdown would cause _notify_connecting more than once if there was still a connection attempt pending. Does ponyup depend on that behaviour?

Reply:
we will go through this path for every happy eyeball attempt so say no ip6 is available but ip4 is.

we will connect via ip4 but when ip6 fails (if it fails after ip4 succeeds) this will notify that the connection failed. that's not what we want for logic here.